userspace/falco/application.h

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The falco "app" will eventually replace the monolithic code in
+// falco.cpp. We expect it will be responsible for the following:
+//  - Parsing/validating command line options
+//  - Parsing/validing falco config
+//  - Initialize prerequsites (inspector, falco engine, webserver, etc)
+//  - Loading plugins
+//  - Loading/validating rules
+//  - Command/subcommand execution (e.g. --list/--list-fields, or
+//    nothing specified to run "main" loop)
+
+// For now, it is only responsible for command line options.
+#pragma once
+
+#include "app_cmdline_options.h"
+
+#include <string>
+
+namespace falco {
+namespace app {
```

Comment:
This is a matter of taste, but if there's not specific need for this namespace pattern I would prefer having a `falco::application` and `falco::cmdline_options` instead, which is less verbose. And since we are here, I think it would be worthy to be consistent with it for the falco engine and other parts of the code too. This would lead to a better naming consistency and less verbose code in each part of the codebase. By that, I mean some renamings like:
- `falco_engine` -> `falco::engine`
- `falco_formats` -> `falco::formats`
- `falco_outputs` -> `falco::outputs`
- ...

This might need be addressed in future PRs, but let's have a discussion about this!

Reply:
I actually like having the application-related objects (application and cmdline_options for now, more to come later) in their own namespace. For example the engine probably *won't* end up below the application-related namespace because it's not really responsible for setting up and managing falco program behavior. I'm not sure formats and outputs would either, although that is less clear to me--it might be a new object that manages formatting/outputs and wraps the falco::formats/falco::outputs objects.

I did make it `falco::app` to keep it concise.